Remove GS Collections

vborisoff opened the pull request:
Going forward the Eclipse collections will be supported.

mohrezaei commented:
you have to rebase this.  "This branch is 1 commit ahead, 3 commits behind goldmansachs:master. "

vborisoff commented:
thanks for pointing out. the latest has:
1. I have addressed the branch merging issues.
2. updates the maven dependency to support https
3. removed misleading info about 1.6
